src/settings.js

```diff
@@ -877,3 +877,8 @@ var MEM_INIT_IN_WASM = 0; // for internal use only
 
 var SUPPORT_BASE64_EMBEDDING = 0; // If set to 1, src/base64Utils.js will be included in the bundle.
                                   // This is set internally when needed (SINGLE_FILE)
+
+var USE_PRINTCHARS_FOR_STDOUT = 1; // If set to 0, printf, puts and other stdio output functions will
```

Comment:
Oh hmm, the original motivation was to be able to avoid having to have a preprocessor defines like this at all.

However it seems that it is too difficult to avoid. Perhaps we should rename this to `var UNBUFFERED_STDIO = 1;` instead to convey what the intent between these two sets of functions is.

Although, I'm now thinking if we are in a scenario where we do want to use a `-s` option for this, then perhaps we might be able to do a full breaking change by overloading `Module.printChars` to just be `Module.print`, which reads much nicer(?)

Reply:
Hmm, my slight concern with UNBUFFERED_STDIO is it is still buffering in musl, so you have to call fflush() to get a string printed without a line-terminator.  Would this be confusing?

Not that I have any better suggestions - perhaps PRINT_PARTIAL_LINES or reverse it and have PRINT_FULL_LINES_ONLY=0?

+1 for overriding Module.print though.

Reply:
On reflection your alternative suggestion of UNBUFFERED_PRINT will probably be fine - the comment describing it in settings.js can make it clear that it doesn't turn off libc stdio buffering.